selene/schema/results/fields.py

```diff
@@ -58,21 +70,77 @@ def resolve_details(root, _info):
         return details.findall('detail')
 
 
+class ResultNVT(ScanConfigNVT):
+    """NVT to which result applies."""
+
+    class Meta:
+        # default resolver is not inherited. Must be declared
+        default_resolver = text_resolver
+
+    version = graphene.String(description='Version of the NVT used in the scan')
+
+    def resolve_version(root, _info):
+        return get_text_from_element(root, 'version')
+
+
+class ResultCVE(graphene.ObjectType):
+    """CVE to which result applies."""
+
+    oid = graphene.String(name='id', description='ID of the CVE')
+    severity = graphene.Field(
+        SeverityType, description='Severity of the CVE result.'
+    )
+
+    def resolve_oid(root, _info):
+        return root.get('oid')
+
+    def resolve_severity(root, _info):
+        return get_text_from_element(root, 'cvss_base')
+
+
+class ResultInformation(graphene.Union):
+    """NVT or CVE to which the result applies."""
+
+    class Meta:
+        types = (ResultNVT, ResultCVE)
+
+    @classmethod
+    def resolve_type(cls, instance, info):
+        info_type = get_text_from_element(instance, 'type')
+
+        if info_type == 'nvt':
+            return ResultNVT
+        else:
+            return ResultCVE
+
+
+class ResultType(graphene.Enum):
+    CVE = 'CVE'
+    NVT = 'NVT'
+
+
 class QoD(graphene.ObjectType):
-    value = graphene.Int()
-    type = graphene.String()
+    value = graphene.Int(description='The numeric QoD value.')
+    qod_type = graphene.String(name="type", description='The QoD type.')
```

Comment:
I've learned yesterday that there are specific values for a qod type. Therefore we should change this field into an enum in another PR later.